llvm/lib/Object/DXContainer.cpp

```diff
@@ -279,14 +270,13 @@ Error DirectX::RootSignature::parse(StringRef Data) {
       support::endian::read<uint32_t, llvm::endianness::little>(Current);
   Current += sizeof(uint32_t);
 
-  uint32_t FValue =
-      support::endian::read<uint32_t, llvm::endianness::little>(Current);
+  Flags = support::endian::read<uint32_t, llvm::endianness::little>(Current);
   Current += sizeof(uint32_t);
 
-  if (!dxbc::RootSignatureValidations::isValidRootFlag(FValue))
-    return validationFailed("unsupported root signature flag value read: " +
-                            llvm::Twine(FValue));
-  Flags = FValue;
+  assert(Current == Begin + RootParametersOffset);
```

Comment:
```suggestion
  assert(Current == Data.begin()+ RootParametersOffset);
```

That way we don't need an otherwise unused `Begin` variable.  I'm surprised the compiler doesn't complain about unused variables when asserts are compiled out.

Although - see my [other comment](https://github.com/llvm/llvm-project/pull/127840/files/f0e6a46f88f9090853637971e652f7a3e905ae26#r2038532814).  I don't think it is safe for us to make this assumption about the parameters immediately following the header.